relayer/provider/cosmos/query.go

```diff
@@ -59,7 +72,32 @@ func (cc *CosmosProvider) QueryTxs(ctx context.Context, page, limit int, events
 	if err != nil {
 		return nil, err
 	}
-	return res.Txs, nil
+
+	var txResps []*provider.RelayerTxResponse
+	for _, tx := range res.Txs {
+		events := parseEventsFromResponseDeliverTx(tx.TxResult)
+		txResps = append(txResps, &provider.RelayerTxResponse{
+			Height: tx.Height,
+			TxHash: string(tx.Hash),
+			Code:   tx.TxResult.Code,
+			Data:   string(tx.TxResult.Data),
+			Events: events,
+		})
+	}
+	return txResps, nil
+}
+
+// parseEventsFromResponseDeliverTx parses the events from a ResponseDeliverTx and builds a map
+// where the keys are equal to event.Type+"."+attribute.Key and the values are the attribute.Value.
+func parseEventsFromResponseDeliverTx(resp abci.ResponseDeliverTx) (events map[string]string) {
+	events = make(map[string]string, 1)
```

Comment:
I'm not sure where the 1 comes from. I don't think we can pre-size the map correctly here without iterating first, so maybe the next best guess would be `events = make(map[string]string, len(resp.Events))` since it would reasonable to assume there is at least one attribute per event.